Automattic/pocket-casts-ios
# Sleep Timer: add the option to select the number of episodes

Adds the option for the user to configure the number of episodes for Sleep Timer.

## To test

Go to Profile > Settings > Beta Features > enable `tracksLogging`.

1. Add a bunch of episodes to your Up Next queue
2. Play an episodes
3. Open the full player > tap Sleep Timer (zZz)
4. Use the + / - button on the penultimate row to change the number of episodes
5. ✅ The label should update accordingly
6. Change it to 3 and tap "In 3 episodes"
7. ✅ Ensure `🔵 Tracked: player_sleep_timer_enabled ["number_of_episodes": 3, "time": "end_of_episode"]` is tracked
8. Tap Sleep Timer (zZz) again
9. ✅ You should see "Sleeping in 3 episodes"
10. Move the scrubber to the end
11. Tap Sleep Timer (zZz) again
12. ✅ You should see "Sleeping in 2 episodes"
13. Move the scrubber to the end
11. Tap Sleep Timer (zZz) again
12. ✅ You should see "Sleeping at the end of the current episode"
13. Move the scrubber to the end
14. ✅ Playback should pause
15. Tap play again
16. ✅ Once playback starts, sleep timer should restart for 3 episodes
17. ✅ Ensure `🔵 Tracked: player_sleep_timer_restarted ["number_of_episodes": 3, "time": "end_of_episode"]` is tracked

### Extending

1. Play an episode
2. Open the full player > tap Sleep Timer (zZz)
3. Choose 30 minutes
4. Tap Sleep Timer (zZz) again
5. Tap "In 3 episodes"
6. ✅ Ensure `🔵 Tracked: player_sleep_timer_extended ["number_of_episodes": 3, "amount": "end_of_episode"]` is tracked
7. ✅ The screen should change to "Sleeping in 3 episodes"

## Checklist
